### PR TITLE
Loki: Remove empty annotations tags

### LINKED ISSUE #32358: [Loki] Labels with empty values are added as annotations
See https://github.com/grafana/grafana/issues/29536 for context

image

> labels with blank values (e.g. "proxy_protocol_addr": "") are still being assigned a tag (see the blank purplish tag third from the top on the far left). I reckon that probably should not be the case. Should I make an issue for that?

### PR DESCRIPTION
**Pull request by conorevans**
Signed-off-by: Conor Evans <coevans@tcd.ie>

**What this PR does / why we need it**: filters labels with empty values.

**Which issue(s) this PR fixes**:

Fixes #32358 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by conorevans** [comment]
>  failed to solve with frontend dockerfile.v0: failed to build LLB: failed to load cache key: failed to copy: httpReaderSeeker: failed open: unexpected status code https://registry-1.docker.io/v2/library/alpine/manifests/sha256:30b2f9f79c929a0905c5bf689806b5e7876a2cbd8befebca7b38bedc7ea694ce: 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

That's why the drone is failing. I can't restart the build myself (I guess I could empty commit). Do you mind kicking it off @ifrost (or @ivanahuckova since you have context on this PR too)?

**Comment by ifrost** [comment]
Restarted the build 👍

File: public/app/plugins/datasource/loki/datasource.ts [hunk 1]
```diff
@@ -511,7 +511,13 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
       const tags: string[] = [];
       for (const field of frame.fields) {
         if (field.labels) {
-          tags.push.apply(tags, [...new Set(Object.values(field.labels).map((label: string) => label.trim()))]);
+          tags.push.apply(tags, [
+            ...new Set(
+              Object.values(field.labels)
+                .filter((label: string) => label !== '')
```
Comment (ifrost) [inline public/app/plugins/datasource/loki/datasource.ts:517]: Question: Would it be worth to swap `filter` and `map` to filter out labels containing only white-space characters?

Comment (conorevans) [reply public/app/plugins/datasource/loki/datasource.ts]: Ah good call, yep! I added a test case for that too.

**Review by ivanahuckova — APPROVED** [review]
Awesome! Thanks for looking into this. 🙏

**Review by ifrost — APPROVED** [review]

**Pull request merged**